pkg/modulereader/resreader.go

```diff
@@ -50,6 +55,47 @@ func (i ModuleInfo) GetOutputsAsMap() map[string]VarInfo {
 	return outputsMap
 }
 
+// GetModuleInfo gathers information about a module at a given source using the
+// tfconfig package. For applicable sources, this function also stages the
+// module contents in a local temp directory and will add required APIs to be
+// enabled for that module.
+func GetModuleInfo(source string, kind string) (ModuleInfo, error) {
+	var modPath string
+	switch {
+	case sourcereader.IsGitPath(source):
+		tmpDir, err := ioutil.TempDir("", "module-*")
+		if err != nil {
+			return ModuleInfo{}, err
+		}
+		modPath = path.Join(tmpDir, "module")
+		sourceReader := sourcereader.Factory(source)
+		if err = sourceReader.GetModule(source, modPath); err != nil {
+			return ModuleInfo{}, fmt.Errorf("failed to clone git module at %s: %v", source, err)
+		}
+
+	case sourcereader.IsEmbeddedPath(source) || sourcereader.IsLocalPath(source):
+		modPath = source
+
+	default:
+		return ModuleInfo{}, fmt.Errorf("Source is not valid: %s", source)
+	}
+
+	reader := Factory(kind)
+	mi, err := reader.GetInfo(modPath)
+
+	// Add API list if source is known
+	if sourcereader.IsEmbeddedPath(source) {
+		mi.RequiredApis = defaultAPIList(modPath)
```

Comment:
I'd modify the comment to read "add APIs required by the module, if known" because it reduces ambiguity.

Reply:
Done